### PR TITLE
[stable/kuberhealthy] Add helm chart flags to disable checks

### PR DESCRIPTION
**Pull request by joshulyne**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

File: stable/kuberhealthy/values.yaml [hunk 1]
```diff
@@ -21,6 +21,22 @@ resources:
     cpu: 400m
     memory: 200Mi
 
+# Disable certain checks by setting enabled to false
+componentStatusCheck:
```
Comment (integrii) [inline stable/kuberhealthy/values.yaml:25]: This isn't bad, but what about a structure like this?

```
checksEnabled:
  daemonSetCheck: true
  dnsStatusCheck: true
  componentStatusCheck: true
  podRestartsCheck: true
  podStatusCheck: true
```

Comment (integrii) [reply stable/kuberhealthy/values.yaml:25]: I suppose your way would be better if we wanted to later on include more options for each check, so I think the way you have it is probably fine - just giving random feedback.

**Comment by integrii** [comment]
The other part of this you need is to add the proper startup flags (or environment variables) to the Kuberhealthy pod spec [here](https://github.com/helm/charts/blob/a9477872922a6b621cdf115df131b17432ac0554/stable/kuberhealthy/templates/deployment.yaml#L69).

Then, of course, you need to add some logic in Kuberhrealthy to allow for the enabling/disabling of checks from environment variables around [here](https://github.com/Comcast/kuberhealthy/blob/b0af1a84606dfd8f634a964bd8bd193b02e186c2/cmd/kuberhealthy/main.go#L44) (see `os.Getenv`).

**Comment by integrii** [comment]
basically the flow is...

helm values -> kubernetes specs -> kuberhealthy environment variable inputs -> kuberhealthy code that turns on/off checks

File: stable/kuberhealthy/values.yaml (hunk 1 above)
Comment (joshulyne) [reply stable/kuberhealthy/values.yaml:25]: Hmm, if we wanted to add more options for each check would we be able to do:
```
anotherOption:
  daemonSetCheck: value
  dnsStatusCheck: value
  componentStatusCheck: value
  podRestartsCheck: value
  podStatusCheck: value
```
??
I guess I'm wondering which structure is better/more appropriate?

**Comment by joshulyne** [comment]
thanks for the comments! will work on next steps :)

File: stable/kuberhealthy/README.md [hunk 2]
```diff
@@ -68,6 +79,41 @@ securityContext: # default container security context
   allowPrivilegeEscalation: false
 ```
 
+The following table lists the configurable parameters of the kuberhealthy chart and their default values.
```
Comment (joshulyne) [inline stable/kuberhealthy/README.md:82]: Added a table here for the configurable parameters, not sure if this is preferred over what we already have!

File: stable/kuberhealthy/values.yaml (hunk 1 above)
Comment (integrii) [reply stable/kuberhealthy/values.yaml:25]: I think the way you did it is fine!  In the future, you would be able to do stuff like this:

```
daemonSetCheck:
  enabled: true
  interval: 30m
  image: image.url.here
```

Forget I said anything!

File: stable/kuberhealthy/README.md (hunk 2 above)
Comment (integrii) [reply stable/kuberhealthy/README.md:82]: Love it!

**Review by integrii — APPROVED** [review]

**Comment by integrii** [comment]
@joshulyne Looks like someone stole your git revision!  Try upping it.

image

**Comment by integrii** [comment]
```
I0518 06:22:19.980] error: timed out waiting for the condition on namespaces/kuberhealthy-presubmit-12863-1129632179798675456-tq0t1mlwpw
I0518 06:22:19.981] Namespace 'kuberhealthy-presubmit-12863-1129632179798675456-tq0t1mlwpw' did not terminate after 180s.
I0518 06:22:20.544] Namespace 'kuberhealthy-presubmit-12863-1129632179798675456-tq0t1mlwpw' did not terminate after 180s.
```

Not sure whats going on.  Could be the test pipeline messing up?

**Pull request closed without merging**